### PR TITLE
[IOTDB-1951] Support aggregation query without value filter in new vector

### PR DESCRIPTION
**Pull request by Cpaulyz**
* adapt new vector about AlignedPath for aggregation
* add IT for vector aggregation

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/query/reader/series/VectorSeriesAggregateReader.java [hunk 1]
```diff
@@ -150,6 +174,11 @@ public BatchData nextPage() throws IOException {
 
   private boolean containedByTimeFilter(Statistics statistics) {
     Filter timeFilter = seriesReader.getTimeFilter();
+    //    if (statistics == null
+    //        || (statistics.getStartTime() == Long.MAX_VALUE
+    //            && statistics.getEndTime() == Long.MIN_VALUE)) {
+    //      return false;
+    //    }
```
Comment (JackieTien97) [inline server/src/main/java/org/apache/iotdb/db/query/reader/series/VectorSeriesAggregateReader.java:181]: delete this.

File: server/src/main/java/org/apache/iotdb/db/query/aggregation/impl/CountAggrResult.java [hunk 2]
```diff
@@ -57,13 +57,18 @@ public void updateResultFromPageData(IBatchDataIterator batchIterator) {
...
       if (batchIterator.currentTime() >= maxBound || batchIterator.currentTime() < minBound) {
         break;
       }
       cnt++;
       batchIterator.next();
     }
+    if (cnt != count) {
+      batchIterator.totalLength();
+      System.out.println("!");
+    }
```
Comment (JackieTien97) [inline server/src/main/java/org/apache/iotdb/db/query/aggregation/impl/CountAggrResult.java:71]: What's this for?

**Review by JackieTien97 — CHANGES_REQUESTED** [review]

**Review by Alima777 — APPROVED** [review]
LGTM

File: server/src/main/java/org/apache/iotdb/db/query/aggregation/impl/CountAggrResult.java [hunk 3]
```diff
@@ -57,6 +57,7 @@ public void updateResultFromPageData(IBatchDataIterator batchIterator) {
   public void updateResultFromPageData(
       IBatchDataIterator batchIterator, long minBound, long maxBound) {
     int cnt = 0;
+    int count = batchIterator.totalLength();
```
Comment (Alima777) [inline server/src/main/java/org/apache/iotdb/db/query/aggregation/impl/CountAggrResult.java:60]: Remove this.

**Review by Alima777 — CHANGES_REQUESTED** [review]

**Pull request merged**